File: llvm-clone-detector/testcases/fortran/factorial.f90
! Test Case 2: Factorial — Fortran implementation
MODULE factorial_mod
  IMPLICIT NONE
CONTAINS

  FUNCTION factorial(n) RESULT(res)
    INTEGER, INTENT(IN) :: n
    INTEGER(KIND=8) :: res
    INTEGER :: i
    res = 1
    DO i = 2, n
      res = res * i
    END DO
  END FUNCTION factorial

  FUNCTION factorial_recursive(n) RESULT(res)
    INTEGER, INTENT(IN) :: n
    INTEGER(KIND=8) :: res
    IF (n <= 1) THEN
      res = 1
    ELSE
      res = n * factorial_recursive(n - 1)
    END IF
  END FUNCTION factorial_recursive

END MODULE factorial_mod

PROGRAM main
  USE factorial_mod
  IMPLICIT NONE
  INTEGER :: i
  DO i = 0, 10
    WRITE(*,*) i, '! =', factorial(i)
  END DO
END PROGRAM main

File: llvm-clone-detector/testcases/c/factorial.c
/* Test Case 2: Factorial — C implementation */
#include <stdio.h>

long factorial(int n) {
    if (n <= 1) return 1;
    long result = 1;
    for (int i = 2; i <= n; i++) {
        result *= i;
    }
    return result;
}

/* Recursive variant — should NOT match iterative */
long factorial_recursive(int n) {
    if (n <= 1) return 1;
    return n * factorial_recursive(n - 1);
}

int main() {
    for (int i = 0; i <= 10; i++)
        printf("%d! = %ld\n", i, factorial(i));
    return 0;
}
